src/main/webapp/js/webrtc_adaptor.js

```diff
@@ -380,32 +380,45 @@ export class WebRTCAdaptor
 		}
 	}
 
-	publish(streamId, token) 
+	publish(streamId, token, subscriberId, subscriberCode) 
 	{
-		//If it started with playOnly mode and wants to publish now
-		if(this.localStream == null){
-			this.navigatorUserMedia(this.mediaConstraints, (stream => {
-				this.gotStream(stream);
-				var jsCmd = {
+		var jsCmd = null;
+		if(subscriberId !== undefined && subscriberCode !== undefined) {
```

Comment:
I think we should have this if else block because it makes the code hard to read and maintain